### DISCUSSION TITLE
Time for a new Documentation template and re-structure?

### DISCUSSION THREAD

**Discussion by lazarusA** [JuliaPlots/Makie.jl#3222] [Ideas]
Looking at https://lux.csail.mit.edu and the use of Documenter/Literate & VuePress, looks like it might be a good idea to follow this workflow. Allowing a better user experience for the Makie community. At the moment the most challenging issue I think it is the creation of small thumbnail images, we should be able to re-use the current approach. As for an initial template I could do an initial draft/PR. Comments?

**Comment by jkrumbiegel** [comment]
Oh yeah I'm all for a rework of our workflow there and we have had some discussions about it. Currently, Simon favors a custom JSServe approach because it would give a lot of control. I personally care a little less about the Julia side of this than having a polished website with good user experience. And because I think nobody here wants to spend too much time on webdesign it would be good to lean on other mature projects in this space.

I had tried a bit to get Docusaurus going but I couldn't make the image preview sites working easily. I was going against the flow of the system multiple times. I had tried to use Pollen.jl to generate appropriate markdown but this was challenging, especially because Docusaurus uses a react markdown hybrid for the cooler features. They do have some cool things though, and react has a ton of nice frontend components one can in principle tap into.

I don't know anything about vuepress or vitepress yet.

**Comment by SimonDanisch** [comment]
I'd love to keep the discussion at https://github.com/MakieOrg/Makie.jl/issues/3105 ;)
